File: core/feature07.go
package main

import (
	"math"
)

func feature07(dd, aa, ee float64) (rr1, rr2, cc, ss float64) {
	Rad := math.Pi / 180.0
	r := dd / 2.0
	bb := math.Atan(r/aa) / Rad
	uu := 1.0 / math.Sin(bb*Rad)
	vv := 180.0 * math.Sin(bb*Rad)
	rr1 = r * uu
	rr2 = rr1 + ee
	cc = 2.0 * r * uu * math.Sin((vv/3.0)*Rad)
	ss = 2.0 * r * uu * math.Sin(vv*Rad)
	return
}

File: core/feature07.js
function feature07() {
  var parameters = extractParameters(["D", "A", "E"]);
  return feature07_core(parameters);
}

function feature07_core(parameters) {
  if (parameters == null) {
    return null;
  }
  var D = getParameter(parameters, "D");
  var A = getParameter(parameters, "A");
  var E = getParameter(parameters, "E");
  var result = {};
  var Rad = Math.PI / 180.0;
  var r = D / 2.0;
  var B = Math.atan(r / A) / Rad;
  var U = 1.0 / Math.sin(B * Rad);
  var V = 180.0 * Math.sin(B * Rad);
  result["Radius"] = [
    r * U,
    r * U + E,
  ];
  result["C"] = 2.0 * r * U * Math.sin((V / 3.0) * Rad);
  result["S"] = 2.0 * r * U * Math.sin(V * Rad);
  return result;
}
